iron-input.html

```diff
@@ -127,11 +158,46 @@
     _bindValueChanged: function() {
       if (this.value !== this.bindValue) {
         this.value = !(this.bindValue || this.bindValue === 0) ? '' : this.bindValue;
+        this._updateBindValues();
       }
       // manually notify because we don't want to notify until after setting value
       this.fire('bind-value-changed', {value: this.bindValue});
     },
 
+    _bindValueAsNumberChanged: function() {
+      var newValue = this.bindValueAsNumber;
+      if ((this.type === 'number' || this.type === 'date') && this.valueAsNumber !== newValue) {
+        try {
```

Comment:
this try/catch is needed in browsers like safari/firefox, where if we try to set:
```javascript
this.valueAsNumber = "a"; // throws exception on safari/firefox, not in chrome